Foundation/NSString.swift

```diff
@@ -630,6 +630,10 @@ extension NSString {
     }
 
     public var boolValue: Bool {
+#if os(Android)
```

Comment:
This makes it inconsistent from other platform only on Android. What's the reasoning here?

Reply:
For whatever reason the NSString.boolValue implementation crashes and it exercises quite a bit of code including charsets which is where the problem most likely lies. I put this in as a stop-gap to keep moving. If you ask me this is what the implementation should be anyway. The Scanner/Charset version seems to make a bit of a meal of it.

Reply:
The semantics do not diverge. The Android code is exactly equivalent to the the existing implementation.